### PR TITLE
Add daterange for data-collector

### PR DESCRIPTION
**Pull request by valouvaliavlo**

### CODE REVIEW

**Review by Herklos — APPROVED** [review]

**Comment by Herklos** [comment]
Can you add the change from https://github.com/Drakkar-Software/OctoBot-Tentacles/pull/486/files#r656593215 please? Thanks!

**Comment by valouvaliavlo** [comment]
Done

**Comment by valouvaliavlo** [comment]
Mayeb we can add `end_timestamp` and `start_timestamp` to sql description to differentiate full and not full data file

File: octobot_backtesting/collectors/exchanges/exchange_collector.py [hunk 1]
```diff
@@ -35,13 +35,15 @@ class ExchangeDataCollector(data_collector.DataCollector):
...
     def __init__(self, config, exchange_name, tentacles_setup_config, symbols, time_frames, use_all_available_timeframes=False,
-                 data_format=enums.DataFormats.REGULAR_COLLECTOR_DATA):
+                 data_format=enums.DataFormats.REGULAR_COLLECTOR_DATA, start_timestamp=None, end_timestamp=None):
         super().__init__(config, data_format=data_format)
         self.exchange_name = exchange_name
         self.tentacles_setup_config = tentacles_setup_config
         self.symbols = symbols if symbols else []
         self.time_frames = time_frames if time_frames else []
         self.use_all_available_timeframes = use_all_available_timeframes
+        self.start_timestamp = start_timestamp
```
Comment (Herklos) [inline octobot_backtesting/collectors/exchanges/exchange_collector.py:45]: Can you also add it to the .pxd file please ?

**Comment by Herklos** [comment]
> Mayeb we can add `end_timestamp` and `start_timestamp` to sql description to differentiate full and not full data file

Good idea :+1:

**Comment by valouvaliavlo** [comment]
`self.config` in `ExchangeDataCollector` is used ? Do you want i set them in ?

**Comment by valouvaliavlo** [comment]
Change `ExchangeDataCollector` to version `1.1` ?

File: tests/data/test_database.py [hunk 2]
```diff
@@ -110,6 +110,11 @@ async def test_select_min():
         assert await database.select_min(enums.ExchangeDataTables.OHLCV, ["timestamp"], ["symbol"], time_frame="1h") == \
             [(1587945600, "ETH/BTC")]
 
+async def test_select_count():
+    async with get_database() as database:
+        assert await database.select_count(enums.ExchangeDataTables.OHLCV, ["*"]) == [(6531,)]
+        assert await database.select_count(enums.ExchangeDataTables.OHLCV, ["*"], time_frame="1h") == [(500,)]
+        assert await database.select_count(enums.ExchangeDataTables.OHLCV, ["*"], time_frame="1M") == [(35,)]
```
Comment (GuillaumeDSM) [inline tests/data/test_database.py:117]: ❤️

File: octobot_backtesting/collectors/data_collector.py [hunk 3]
```diff
@@ -61,10 +62,14 @@ def _ensure_file_path(self):
...
         self.file_path = path.join(self.path, self.file_name) if self.path else self.file_name
+        self.temp_file_path = self.file_path + constants.BACKTESTING_DATA_FILE_TEMP_EXT
 
     def create_database(self) -> None:
         if not self.database:
-            self.database = data.DataBase(self.file_path)
+            self.database = data.DataBase(self.temp_file_path)
+    
+    def finalize_database(self):
+        os.rename(self.temp_file_path, self.file_path)
```
Comment (GuillaumeDSM) [inline octobot_backtesting/collectors/data_collector.py:72]: 👍

**Review by GuillaumeDSM — APPROVED** [review]

**Pull request merged**